src/prioritize-features.py

```diff
@@ -0,0 +1,173 @@
+the label will be ranked above all other Blocker features. All Critical features in the label will
+be ranked above all other Critical features. All Major features etc..)
+
+"""
+
+import argparse
+import os
+import sys
+
+import click
+import jira
+
+PRIORITY = [
+    "Undefined",
+    "Minor",
+    "Normal",
+    "Major",
+    "Critical",
+    "Blocker",
+]
+
+DRY_RUN = False
+
+
+@click.command(
+    help=__doc__,
+)
+@click.option(
+    "--dry-run",
+    help="Do not update issues.",
+    is_flag=True,
+)
+@click.option(
+    "-l",
+    "--label",
+    help="Label we are prioritizing",
+    required=True,
+)
+@click.option(
+    "-p",
+    "--project-id",
+    help="Project that we are prioritizing in",
+    required=True,
+)
+@click.option(
+    "-t",
+    "--token",
+    help="JIRA personal access token",
+    default=os.environ.get("JIRA_TOKEN"),
+    required=True,
+)
+@click.option(
+    "-u",
+    "--url",
+    help="JIRA URL",
+    default=os.environ.get("JIRA_URL", "https://issues.redhat.com"),
+)
+def main(dry_run: bool, label: str, project_id: str, token: str, url: str) -> None:
+    global DRY_RUN
+    DRY_RUN = dry_run
+    jira_client = jira.client.JIRA(server=url, token_auth=token)
+
+    all_fields = jira_client.fields()
+    jira_name_map = {field["name"]: field["id"] for field in all_fields}
+    rank_key = jira_name_map["Rank"]
+
+    config = {
+        "issues": ["Feature"],
+        "rank_key": rank_key,
+    }
+
+    for issue_type in config["issues"]:
+        process_type(jira_client, label, project_id, issue_type, config)
+    print("Done.")
+
+
+def process_type(
+    jira_client: jira.client.JIRA,
+    label: str,
+    project_id: str,
+    issue_type: str,
+    config: dict,
+) -> None:
+    print(f"\n\n## Processing {issue_type}")
+
+    priorities = PRIORITY
+    issues = get_issues(jira_client, label, issue_type)
+    top_issues = get_highest_ranked_issues(
+        jira_client, priorities, project_id, issue_type
+    )
+
+    for issue in issues:
+        print(f"### {issue.key}")
+        context = {
+            "updates": [],
+            "jira_client": jira_client,
+            "rank_key": config["rank_key"],
+        }
+        check_rank(issue, context, top_issues)
+        add_comment(issue, context)
+
+
+def get_highest_ranked_issues(
+    jira_client: jira.client.JIRA,
+    priorities: list[str],
+    project_id: str,
+    issue_type: str,
+) -> dict:
+    """Return a dict of the highest ranked issues with each priority"""
+    results = {}
+    for priority in priorities:
+        query = f"priority={priority} AND project={project_id} AND type={issue_type} ORDER BY Rank ASC"
+        issues = jira_client.search_issues(query, maxResults=1)
+        if not issues:
+            print(f"No {issue_type} found via query: {query}")
+            sys.exit(1)
+        results[priority] = issues[0]
+    return results
+
+
+def get_issues(
```

Comment:
1. If for some reason 2 PM Projects use the same label, this is going to impact people that do not want to be impacted. I would change the behavior so that the reranking happens only within a project. The other script can then be used to cascade the priorities/ranking in individual projects. This can either be done manually, or you can list the impacted projects as you go through the features and then trigger the script for each one of them as the last step.
2. If you agree with the above changes, you might be able to tweak the `src/utils/jira:query_issues()` to get the  Features you want.

You might be able to upgrade `src/utils/jira.py:query_issues()` to do what you want.

Reply:
Good call, thanks!